app/src/main/java/com/codingblocks/cbonlineapp/commons/LeaderboardDiffCallback.kt

```diff
@@ -0,0 +1,28 @@
+package com.codingblocks.cbonlineapp.commons
+
+import androidx.recyclerview.widget.DiffUtil
+import com.codingblocks.onlineapi.models.Leaderboard
+
+/**
+ * The DiffUtil ItemCallback class for the [Leaderboard] model class.
+ * This enables proper diffing of items in Recycler Views using [DiffUtil]
+ */
+class LeaderboardDiffCallback : DiffUtil.ItemCallback<Leaderboard>() {
+
+    /**
+     * return true if the contents of both items are same
+     */
+    override fun areContentsTheSame(oldItem: Leaderboard, newItem: Leaderboard): Boolean {
+        return oldItem.userName == newItem.userName &&
```

Comment:
why not just check the entire object instead of checking every attribute ?

Reply:
I tried that first which gave this warning.
```Suspicious equality check: equals() is not implemented in Leaderboard```
I wonder why this warning is not there in case of Notification in NotificationDiffCallback.

Reply:
Most likely because leaderboard is not a data class right ?

Reply:
Convert leaderboard class into a data.class

Reply:
Done.